### PR TITLE
Import checks the file has all the required columns

### PR DESCRIPTION
**Pull request by aorin**
https://www.pivotaltracker.com/story/show/179571097
 https://177999735.dev.laji.fi/vihko/tools/import

Import checks that the file has all the required columns. If the file has a split column part (e.g. Alku@dd - Yleinen keruutapahtuma), it checks that it has all the remaining parts (Alku@mm - Yleinen keruutapahtuma, Alku@yyyy - Yleinen keruutapahtuma)

### CODE REVIEW

File: projects/laji/src/app/shared-modules/spreadsheet/service/spreadsheet.service.ts [hunk 1]
```diff
@@ -289,18 +303,46 @@ export class SpreadsheetService {
...
+        return groups[group];
+      }),
+      errors
+    };
+  }
+
+  private checkCombinedHasRequiredColumns(combines: IColCombine[], groupType: GroupTypeEnum, errors: string[]): IColCombine[] {
+    const requiredKeys = (groupType === GroupTypeEnum.date ? [GeneratorService.splitDate.dd, GeneratorService.splitDate.mm, GeneratorService.splitDate.yyyy] :
+      [GeneratorService.splitCoordinate.E, GeneratorService.splitCoordinate.N, GeneratorService.splitCoordinate.system]
+    );
```
Comment (olzraiti) [inline projects/laji/src/app/shared-modules/spreadsheet/service/spreadsheet.service.ts:330]: Quite long line, could the ternary operation be splitted to multiple lines like this (IMO the most readable option anyway)

```
question
  ? yes
  : no
```

We should add a max line length rule to linting...

File: projects/laji/src/app/shared-modules/spreadsheet/service/spreadsheet.service.ts [hunk 2]
```diff
@@ -289,18 +303,46 @@ export class SpreadsheetService {
...
+
+    const fields = combines.filter(col => col.groupType === groupType);
+    if (fields.length > 0) {
+      const keys = fields.map(col => col.type);
+      let missing = requiredKeys.filter(key => !keys.includes(key));
+      if (missing.includes(GeneratorService.splitCoordinate.system) && keys.includes(GeneratorService.splitCoordinate.sys)) {
+        missing = missing.filter(val => val !== GeneratorService.splitCoordinate.system);
+      }
+      if (missing.length > 0) {
+        errors.push(this.translateService.instant('excel.import.error.missingSplitColumn', {field: fields[0].field, column: missing.join(', ')}));
```
Comment (olzraiti) [inline projects/laji/src/app/shared-modules/spreadsheet/service/spreadsheet.service.ts:340]: Also a quite long line, please split into multiple lines.

**Review by olzraiti — CHANGES_REQUESTED** [review]
Looking good, just minor line length nits

**Comment by aorin** [comment]
Those should be fixed now

**Review by olzraiti — APPROVED** [review]
Thanks!

**Pull request merged**